test/utils.h

```diff
@@ -382,6 +383,21 @@ class ContextCudnnTF32Disabled {
   bool flag_;
 };
 
+inline bool maybeClearAllocator(int64_t max_bytes = ((int64_t)1 << 32)) {
+  // check used memory and empty allocator cache if above a set threshold
+  auto allocator = c10::cuda::CUDACachingAllocator::get();
+  if (allocator->initialized()) {
+    auto device_stats = allocator->getDeviceStats(0);
```

Comment:
Yeah that would be nice but I couldn't figure out how to find either the last used device or a list of devices handled by the allocator: https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.h#L199.

Reply:
Sorry for not being clear earlier, you can directly call `cudaGetDevice()`.. Since PyTorch also uses runtime API for that under the hood.

There's some behavior change on cuda12 I think. So if you want to play nice along with other pytorch mechanics, you might want to use https://github.com/pytorch/pytorch/blob/80dd847b62ec526827d363bf9b07e52ef7499d38/c10/cuda/CUDAFunctions.h#L40

Reply:
I'd like to guard it with an `#if` to try and use `GetDevice` but not sure what the condition should be for c10. Looking into it...

Reply:
Found https://pytorch.org/cppdocs/notes/versioning.html. I'll guard with this for PT>2.0.1 and fall back to `cudaGetDevice`.